rmw_connextdds_common/src/common/rmw_impl.cpp

```diff
@@ -744,6 +744,7 @@ RMW_Connext_Publisher::create(
       topic_mapping_i =
         std::make_unique<RMW_Connext_TopicMappingInstance>(
         dp, app_topic, app_type, net_topic, net_type, transform_plugin);
+      ctx->config.notify_type_use(topic_mapping_i->get_net_typecode());
```

Comment:
After reviewing the code in `DDS_DomainParticipant_register_type()`, I think there is no need to rely on the TypeCode for counting references, but we can instead rely just on the type name. The successful call to `DDS_DomainParticipant_register_type()` (which is performed within the constructor of `RMW_Connext_TopicMappingInstance`) guarantees that the `transform_plugin` either uses the same type as a previously registered type, or it registered a new type. We should also be aware that if `transform_plugin` is `nullptr` then we don't need to keep track of the reference counting.

I suggest updating this block to the following:

```suggestion
      if (nullptr != transform_plugin) {
        ctx->config.on_network_type_registered(net_type);
      }
```